src/Standard.Licensing/Security/Cryptography/KeyFactory.cs

```diff
@@ -43,14 +48,33 @@ internal static class KeyFactory
         /// <returns>The encrypted private key.</returns>
         public static string ToEncryptedPrivateKeyString(AsymmetricKeyParameter key, string passPhrase)
         {
-            var salt = new byte[16];
-            var secureRandom = SecureRandom.GetInstance("SHA256PRNG");
-            secureRandom.SetSeed(secureRandom.GenerateSeed(16)); //See Bug #135
-            secureRandom.NextBytes(salt);
+            // Rent a buffer for the salt (16 bytes for the salt)
+            var salt = ArrayPool<byte>.Shared.Rent(16);
 
-            return
-                Convert.ToBase64String(PrivateKeyFactory.EncryptKey(keyEncryptionAlgorithm, passPhrase.ToCharArray(),
-                                                                    salt, 10, key));
+            try
+            {
+#if NET6_0_OR_GREATER
+                // Use built-in cryptography (e.g., RNGCryptoServiceProvider) in .NET 6.0+
+                using (var rng = RandomNumberGenerator.Create())
+                {
+                    rng.GetBytes(salt, 0, 16);
+                }
```

Comment:
You could tighten this line up a bit and get rid of the using with a change to using the static `Fill()` method:
`RandomNumberGenerator.Fill(salt.AsSpan(0,16));`